pkg/provider/containergroup_to_pod.go

```diff
@@ -5,109 +5,72 @@ Licensed under the Apache 2.0 license.
 package provider
 
 import (
-	"fmt"
+	"time"
 
 	azaci "github.com/Azure/azure-sdk-for-go/services/containerinstance/mgmt/2021-10-01/containerinstance"
+	"github.com/pkg/errors"
+	"github.com/virtual-kubelet/azure-aci/pkg/tests"
 	v1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/types"
 )
 
-func containerGroupToPod(cg *azaci.ContainerGroup) (*v1.Pod, error) {
-	_, creationTime := getACIResourceMetaFromContainerGroup(cg)
+var (
+	emptyStr = ""
+)
 
-	containers := make([]v1.Container, 0, len(*cg.Containers))
-	containersList := *cg.Containers
-	for i := range containersList {
-		container := &v1.Container{
-			Name:  *containersList[i].Name,
-			Image: *containersList[i].Image,
-			Resources: v1.ResourceRequirements{
-				Requests: v1.ResourceList{
-					v1.ResourceCPU:    resource.MustParse(fmt.Sprintf("%g", *containersList[i].Resources.Requests.CPU)),
-					v1.ResourceMemory: resource.MustParse(fmt.Sprintf("%gG", *containersList[i].Resources.Requests.MemoryInGB)),
-				},
-			},
-		}
-		if containersList[i].Command != nil {
-			container.Command = *containersList[i].Command
+func (p *ACIProvider) containerGroupToPod(cg *azaci.ContainerGroup) (*v1.Pod, error) {
+	//cg is validated
+	pod := v1.Pod{}
+	podList := p.resourceManager.GetPods()
```

Comment:
I think the logic of containerGroupToPod should be:

1) Get Pod name and namespace from cg. Then query the apiserver to get Pod object.
2) Make a deepcopy of the returned Pod object. 
3) Updating the podCopy.Status with the result of `p.getPodStatusFromContainerGroup(cg)`
4) return podCopy.